signer/bls/cerberus/cerberus.go

```
@@ -0,0 +1,129 @@
+package cerberus
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"log"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/insecure"
+
+	sdkBls "github.com/Layr-Labs/eigensdk-go/crypto/bls"
+	"github.com/Layr-Labs/eigensdk-go/signer/bls/types"
+	"github.com/consensys/gnark-crypto/ecc/bn254"
+
+	v1 "github.com/Layr-Labs/cerberus-api/pkg/api/v1"
+)
+
+type Config struct {
+	URL          string
+	PublicKeyHex string
+
+	// Optional: in case if your signer uses local keystore
+	Password string
+
+	EnableTLS       bool
+	TLSCertFilePath string
+
+	SigningTimeout time.Duration
+}
+
+type Signer struct {
+	signerClient v1.SignerClient
+	kmsClient    v1.KeyManagerClient
+	pubKeyHex    string
+	password     string
+}
+
+func New(cfg Config) (Signer, error) {
+	opts := make([]grpc.DialOption, 0)
+	if cfg.EnableTLS {
+		creds, err := credentials.NewClientTLSFromFile(cfg.TLSCertFilePath, "")
+		if err != nil {
+			log.Fatalf("could not load tls cert: %s", err)
+		}
+		opts = append(opts, grpc.WithTransportCredentials(creds))
+	} else {
+		opts = append(opts, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	}
+
+	conn, err := grpc.NewClient(cfg.URL, opts...)
+	if err != nil {
+		log.Fatalf("did not connect: %v", err)
```

any reason why it's crashing instead of returning error?

---

ah good catch, I will update.

---

2357268d8a189b99301d7d121135e6a84ea322ad